cpp/src/arrow/flight/flight_internals_test.cc

```diff
@@ -177,6 +175,25 @@ TEST(FlightTypes, RoundtripStatus) {
   ASSERT_THAT(status.message(), ::testing::HasSubstr("Sentinel"));
 }
 
+ARROW_SUPPRESS_DEPRECATION_WARNING
+TEST(FlightTypes, DeprecatedLocationConstruction) {
+  Location location;
+  ASSERT_NOT_OK(Location::Parse("This is not an URI", &location));
```

Comment:
Can we use a more restrictive test here, e.g.:
```suggestion
  ASSERT_RAISES(Invalid, Location::Parse("This is not an URI", &location));
```